rust/oak/src/grpc/mod.rs

```diff
@@ -22,29 +22,56 @@ use protobuf::{Message, ProtobufEnum};
 /// Result type that uses a [`proto::status::Status`] type for error values.
 pub type Result<T> = std::result::Result<T, proto::status::Status>;
 
-/// Trait to allow repeated writing of responses for server-streaming gRPC methods.
-pub trait ResponseWriter<T: protobuf::Message> {
-    fn write(&mut self, rsp: T);
+/// Channel-holding object that encapsulates response messages into
+/// `GrpcResponse` wrapper messages and writes serialized versions to a send
+///  channel.
+pub struct ChannelResponseWriter {
+    channel: crate::io::Channel,
+    stream_id: i32,
 }
 
-/// Implementation of [`ResponseWriter`] that encapsulates response messages
-/// into `GrpcResponse` wrapper messages and writes serialized versions to a
-/// (mutably borrowed) send channel.
-pub struct ChannelResponseWriter<'a> {
-    pub channel: &'a mut crate::io::Channel,
-}
+impl ChannelResponseWriter {
+    pub fn new(out_handle: crate::WriteHandle, stream_id: i32) -> Self {
+        ChannelResponseWriter {
+            channel: crate::io::Channel::new(out_handle),
+            stream_id,
+        }
+    }
 
-impl<'a, T> ResponseWriter<T> for ChannelResponseWriter<'a>
-where
-    T: protobuf::Message,
-{
-    fn write(&mut self, rsp: T) {
-        // Put the serialized response into a GrpcResponse message wrapper.
+    /// Write out a gRPC response and optionally close out the method
+    /// invocation.
+    pub fn write<T: protobuf::Message>(&mut self, rsp: T, close: bool) {
```

Comment:
The `close` bool is not very readable in calls. How about either two different methods `write_last` and `write_more` (or something better), or define an enum `WriteMode::{KeepOpen,Close}` or `Write::{More,Last}`?

Reply:
Done (went for the `enum`)